### ISSUE TITLE
Combination of no_mangle and used attribute leads to mangled symbol name

### ISSUE THREAD

**Issue by Tiwalun** [rust-lang/rust#99732]
I tried this code:

```rust
#[no_mangle]
#[used]
pub static exported_static: u32 = 10;
```

This is compiled into a binary. I expect that the symbol is part of the binary, and that the name is not mangled, due to the `no_mangled` attribute. Example nm output of a correct case:

```
...
00000240 T exported_static
...
```

Instead, the symbol name gets mangled:

```
...
00000240 t _ZN18hifive_flashloader15exported_static17hef539998df4d12acE
...
```

However, if I remove the `#[used]` attribute, then it works correctly:
```rust
#[no_mangle]
pub static exported_static: u32 = 10;
```

### Meta

This only occurs on the nightly compiler:

 ```
rustc 1.64.0-nightly (7fe022f5a 2022-07-24)```

 On stable rustc 1.62.1, the symbol does not get mangled.

**Comment by jannic** [comment]
Probably already fixed with commit 6cfd471ffe997c56848d27e688dd8bd30f18412a by @thomcc

**Comment by Tiwalun** [comment]
I can confirm that it's fixed in `rustc 1.64.0-nightly (6dbae3ad1 2022-07-25)`.
